modules/identity/src/Volo.Abp.Identity.Application/Volo/Abp/Identity/ProfileAppService.cs

```diff
@@ -56,6 +61,20 @@ public virtual async Task<ProfileDto> UpdateAsync(UpdateProfileDto input)
         public virtual async Task ChangePasswordAsync(ChangePasswordInput input)
         {
             var currentUser = await UserManager.GetByIdAsync(CurrentUser.GetId());
+
+            if (currentUser.IsExternal)
+            {
+                throw new BusinessException(code: IdentityErrorCodes.ExternalUserPasswordChange);
+            }
+
+            if (currentUser.PasswordHash == null)
+            {
+                (await UserManager.RemovePasswordAsync(currentUser)).CheckErrors();
```

Comment:
Is it really need to `UserManager.RemovePasswordAsync`. I suppose it is not needed because you are already checking if currentUser.PasswordHash == null.